sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala

```diff
@@ -67,6 +68,9 @@ object CollateExpressionBuilder extends ExpressionBuilder {
  */
 case class Collate(child: Expression, collationName: String)
   extends UnaryExpression with ExpectsInputTypes {
+  if (!SQLConf.get.collationEnabled) {
```

Comment:
Can you also disable it in `visitCollate` in `AstBuilder.scala`?

Reply:
+1

Reply:
I firstly did this, but changed it back, since visitCollate calls Collate() and I thought it would be double guard. Do we want to do double guard in this case?